### DISCUSSION TITLE
Module fragmentation

### DISCUSSION THREAD

**Discussion by 6axis** [HALRobotics/Beta#88] [Ideas]
Hello, 

Working with a big amout of targets is impossible to a certain point. It seem like spliting one big heavy module & procedure into multiple ones with a reasonable amount of targets could work. 

Are you considering implementing a fonctionnality that allow user to chose to split big procedure into multiple ones inside a new module ? 
 
I don't have a deep understanding of how a robot controller work, so maybe I'm missing something.

Regards

**Comment by sebandraos** [comment]
Hi @6axis ,
Yes, absolutely. We already have the functionality within the Framework, and where the language allows, to split Procedures into multiple "functions" within a program file but we are very aware that there are times when these should really be split into multiple files and loaded automatically as part of an aggregator program. We have a few ideas of how that could be done internally and will try to make something available "soon" to handle this use case. We have some linguistics work lined up for April so hopefully we can fit this feature in at the same time.
